report/chapters/problem_domain_analysis.typ

```diff
@@ -1,12 +1,247 @@
+
+At this stage, the process avoided premature judgment, allowing for the exploration of a wide range of possible class candidates. This creative approach ensured that no relevant entities were discarded too soon, which could later prove useful for the system’s design.
+
+=== Finding Candidates for Events
+After establishing a comprehensive list of class candidates, the next step was to identify potential event candidates. Events represent actions or occurrences within the system that trigger specific behaviors, such as logging an absence or updating a calendar.
+
+To derive event candidates, the dynamic interactions between the classes were considered. For example, the Absence Entry event is triggered when an employee logs an absence, while the Calendar View Update event occurs when the system updates to reflect new absences. More specific events were also explored, such as Absence Update, Absence Notification, and Team Creation, each representing different system activities.
+
+By identifying these events, a clearer understanding of how the system would operate dynamically began to emerge. Each event was mapped to the relevant classes, ensuring that the system could respond appropriately to user actions and data changes.
+
+=== Systematically Evaluating Classes and Events
+After identifying a broad set of class and event candidates, the final step was to systematically evaluate the relationships between them. This involved critically analyzing the roles of each class and the events they participate in. An event table was used for this purpose, mapping classes to the events they are involved in.
+
+The event table allowed for an evaluation of the cohesion and coupling within the system. For example:
+
+- If a class was involved in too many events, it suggested that the class might have too many responsibilities, indicating low cohesion. In such cases, breaking the class into smaller, more focused entities was considered.
+- If an event connected many classes, it indicated high coupling, which could make the system harder to maintain. Such cases were examined closely to ensure that the connections were necessary and not overly complex.
+This systematic evaluation helped refine the class and event candidates, ensuring that the system design would be both maintainable and aligned with the core functionality required by KMD employees. The resulting set of well-defined classes and events formed the foundation of the system’s architecture.
+
+=== Final Event Table
+After several iterations and refinements, the following event table was developed, capturing the core interactions within the KMD Absence Planner:

```

Comment:
After several iterations and refinements

probably not necessary to emphasize the magnitude of work. (bl.a. for at vi ikke kan få hug fra eksamen for at det er ikke er "raffineret nok" xD